indico/modules/events/registration/controllers/api/checkin.py

```diff
@@ -107,3 +107,18 @@ def _process_PATCH(self, checked_in=None, paid=None):
         if paid is not None and paid != self.registration.is_paid and self.registration.price:
             toggle_registration_payment(self.registration, paid=paid)
         return CheckinRegistrationSchema().jsonify(self.registration)
+
+
+class RHCheckinAPIRegistrationUUID(RHCheckinAPIBase):
+    """Get full details for a specific registration from the ticket UUID (checkin secret)."""
+
+    def _process_args(self):
+        ticket_uuid = str(request.view_args['ticket_uuid'])
+        self.registration = (Registration.query
+                             .filter_by(ticket_uuid=ticket_uuid, is_deleted=False)
+                             .first_or_404())
+        self.regform = self.registration.registration_form
+        self.event = self.registration.event
```

Comment:
```python
if self.regform.is_deleted or self.event.is_deleted:
    raise NotFound
```

I don't think this is checked anywhere else...

Reply:
nevermind, event deleted is checked in `RHEventBase`. but for the regform there is no such check anywhere else